test/firtool/blackbox.mlir

```diff
@@ -0,0 +1,62 @@
+// RUN: firtool %s --format=mlir --split-verilog -o=%t
+// RUN: FileCheck %s --check-prefix=VERILOG-TOP < %t/test_mod.sv
+// RUN: FileCheck %s --check-prefix=VERILOG-FOO < %t/magic/blackbox-inline.v
+// RUN: FileCheck %s --check-prefix=VERILOG-HDR < %t/magic/blackbox-inline.svh
+// RUN: FileCheck %s --check-prefix=VERILOG-BAR < %t/magic/blackbox-resource.v
+// RUN: FileCheck %s --check-prefix=VERILOG-GIB < %t/magic/blackbox-path.v
+// RUN: FileCheck %s --check-prefix=LIST-TOP < %t/filelist.f
+// RUN: FileCheck %s --check-prefix=LIST-BLACK-BOX < %t/magic.f
+
+// LIST-TOP: test_mod.sv
+
+// LIST-BLACK-BOX:      magic/blackbox-inline.v
+// LIST-BLACK-BOX-NEXT: magic/blackbox-resource.v
+// LIST-BLACK-BOX-NEXT: magic/blackbox-path.v
+
+firrtl.circuit "test_mod" attributes {annotations = [
+  // Black box processing should honor only the last annotation.
+  {class = "firrtl.transforms.BlackBoxTargetDirAnno", targetDir = "ignore_me_plz"},
+  {class = "firrtl.transforms.BlackBoxTargetDirAnno", targetDir = "magic"},
+  {class = "firrtl.transforms.BlackBoxResourceFileNameAnno", resourceFileName = "definitely_bad.f"},
+  {class = "firrtl.transforms.BlackBoxResourceFileNameAnno", resourceFileName = "magic.f"}
```

Comment:
Ugh... This is the right behavior by my read of the Scala, but I hate this. It means that annotation order becomes part of the API. 😬 😬 😬 Good catch to check this.

Reply:
Yeah that `fold` in the Scala code is magic :mage: